plugins/module_utils/network/sonic/argspec/ospfv2/ospfv2.py

```diff
@@ -0,0 +1,173 @@
+                        'administrative': {'type': 'bool'},
+                        'external_lsa_all': {'type': 'int'},
+                        'external_lsa_connected': {'type': 'int'},
+                        'on_startup': {'type': 'int'},
+                        'router_lsa_all': {'type': 'int'},
+                        'router_lsa_stub': {'type': 'int'}
+                    },
+                    'type': 'dict'
+                },
+                'maximum_paths': {'type': 'int'},
+                'non_passive_interfaces': {
+                    'elements': 'dict',
+                    'options': {
+                        'addresses': {
+                            'elements': 'dict',
+                            'options': {
+                                'address': {'type': 'str'},
+                            },
+                            'type': 'list'
+                        },
+                        'interface': {'required': True, 'type': 'str'}
+                    },
+                    'type': 'list'
+                },
+                'opaque_lsa_capability': {'type': 'bool'},
+                'passive_interfaces': {
+                    'elements': 'dict',
+                    'options': {
+                        'addresses': {
+                            'elements': 'dict',
+                            'options': {
+                                'address': {'type': 'str'},
+                            },
+                            'type': 'list'
+                        },
+                        'interface': {'required': True, 'type': 'str'}
+                    },
+                    'type': 'list'
+                },
+                'redistribute': {
+                    'elements': 'dict',
+                    'options': {
+                        'metric': {'type': 'int'},
+                        'metric_type': {'choices': [1, 2], 'type': 'int'},
+                        'protocol': {
+                            'choices': ['bgp', 'kernel', 'connected', 'static'],
+                            'required': True,
+                            'type': 'str'
+                        },
+                        'route_map': {'type': 'str'}
+                    },
+                    'type': 'list'
+                },
+                'refresh_timer': {'type': 'int'},
+                'rfc1583_compatible': {'type': 'bool'},
+                'router_id': {'type': 'str'},
+                'timers': {
+                    'options': {
+                        'lsa_min_arrival': {'type': 'int'},
+                        'throttle_lsa_all': {'type': 'int'},
+                        'throttle_spf': {
+                            'options': {
+                                'delay_time': {'required': True, 'type': 'int'},
+                                'initial_hold_time': {'required': True, 'type': 'int'},
+                                'maximum_hold_time': {'required': True, 'type': 'int'}
```

Comment:
Since, all options are listed as 'required_together' specifying each option as 'required: True' is not needed.